### DISCUSSION TITLE
Refactoring schema, from Joi.required() to z.any()

### DISCUSSION THREAD

**Discussion by zzhikai** [colinhacks/zod#2037] [Q&A] [answered]
New to Zod as well as Joi, was trying to refactor Joi.required() to z.any() in my schema but I realised that the z.any() behaviour seems to not be 'required' by default, wondering if anyone knows what the Zod equivalent for Joi.required(without any types)?

**Comment by JacobWeisenburger** [comment, marked as answer]
Is this what you are looking for?

https://joi.dev/api/?v=17.7.0#anyrequired---aliases-exist
```ts
const schema = Joi.any().required()
```

```ts
const schema = z.custom( value => value !== undefined )
console.log( schema.parse( 'anything but undefined' ) ) // anything but undefined

const result = schema.safeParse( undefined )
!result.success && console.log( result.error.issues )
// [ { code: 'custom', fatal: undefined, path: [], message: 'Invalid input' } ]
```

If you found my answer satisfactory, please consider supporting me. Even a small amount is greatly appreciated. Thanks friend! 🙏
https://github.com/sponsors/JacobWeisenburger

**Reply by zzhikai** [reply]
Thank you!
